src/secureCreatives.js

```diff
@@ -108,6 +109,13 @@ function handleNativeRequest(reply, data, adObject) {
     logError(`Cannot find ad '${data.adId}' for x-origin event request`);
     return;
   }
+
+  if (!WON_AD_IDS.has(adObject.adId)) {
```

Comment:
`WON_AD_IDS` could be a  WeakMap using adObject as key - otherwise it's unbounded. Not a big deal as I don't think we ever throw `adObject` to the garbage collector, but one day we might.

Reply:
With weakmaps, i'd have to change line 114 to `WON_AD_IDS.set(adObject, adObject)` and honestly i don't like the repetition :) 

Since the complexity is still sublinear (as [written in the ecmaspec](https://262.ecma-international.org/6.0/#sec-weakset-objects)), i preferred to use WeakSet, because it maps more naturally to what we're doing here. What do you think? the only major problem was compatibility with IE, but Prebid already dropped it in version 6.